### ISSUE TITLE
RFE: Silence notification messages logged to the console output when no notifications are enabled

### ISSUE THREAD

**Issue by ryankwilliams** [RedHatQE/teflo#174]
It would be great if teflo could silence some of the messages logged to the console output about notifications. When a teflo user does not use the built in notification plugins for certain actions, it produces a good amount of output by default (info logging level). Would it be possible to make this output smaller or even switch it to (debug logging level) and only print out in (info logging level) if its actually sending a notification?

```
2021-12-06 00:51:13,839 INFO                                TEFLO RUN (START)                               
2021-12-06 00:51:13,840 INFO -------------------------------------------------------------------------------
2021-12-06 00:51:13,840 INFO  * Data Folder           : /home/jenkins/agent/workspace/PipelineSimulator/sol-mock-openshift4.8/.teflo/wxr157zmlg
2021-12-06 00:51:13,840 INFO  * Workspace             : .
2021-12-06 00:51:13,841 INFO  * Log Level             : info
2021-12-06 00:51:13,841 INFO  * Tasks                 : ['orchestrate']
2021-12-06 00:51:13,841 INFO  * Iterate Method        : by_level
2021-12-06 00:51:13,842 INFO  * Scenario              : resource_check
2021-12-06 00:51:13,842 INFO  * Scenario              : provision_localhost
2021-12-06 00:51:13,842 INFO  * Scenario              : orchestrate_stack
2021-12-06 00:51:13,843 INFO  * Scenario              : bare_metal_stack
2021-12-06 00:51:13,843 INFO -------------------------------------------------------------------------------

2021-12-06 00:51:13,843 INFO ..................................................
2021-12-06 00:51:13,844 INFO [32m'resource_check' is running from the scenario file: resource_check_results.yml[0m
2021-12-06 00:51:13,844 INFO ..................................................
2021-12-06 00:51:13,844 INFO Sending out any notifications that are registered.
2021-12-06 00:51:13,845 INFO ..................................................
2021-12-06 00:51:13,846 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:13,846 WARNING ... no tasks to be executed ...
2021-12-06 00:51:13,846 INFO  * Task    : orchestrate
2021-12-06 00:51:13,847 INFO ..................................................
2021-12-06 00:51:13,897 INFO Starting tasks on pipeline: orchestrate
2021-12-06 00:51:13,897 WARNING ... no tasks to be executed ...
2021-12-06 00:51:13,898 INFO ..................................................
2021-12-06 00:51:13,898 INFO Sending out any notifications that are registered.
2021-12-06 00:51:13,899 INFO ..................................................
2021-12-06 00:51:13,899 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:13,900 WARNING ... no tasks to be executed ...
2021-12-06 00:51:13,900 INFO ..................................................
2021-12-06 00:51:13,900 INFO [32m'provision_localhost' is running from the scenario file: provision_localhost_results.yml[0m
2021-12-06 00:51:13,901 INFO ..................................................
2021-12-06 00:51:13,901 INFO Sending out any notifications that are registered.
2021-12-06 00:51:13,902 INFO ..................................................
2021-12-06 00:51:13,902 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:13,903 WARNING ... no tasks to be executed ...
2021-12-06 00:51:13,903 INFO  * Task    : orchestrate
2021-12-06 00:51:13,904 INFO ..................................................
2021-12-06 00:51:13,904 INFO Starting tasks on pipeline: orchestrate
2021-12-06 00:51:13,905 WARNING ... no tasks to be executed ...
2021-12-06 00:51:13,905 INFO ..................................................
2021-12-06 00:51:13,905 INFO Sending out any notifications that are registered.
2021-12-06 00:51:13,906 INFO ..................................................
2021-12-06 00:51:13,906 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:13,907 WARNING ... no tasks to be executed ...
2021-12-06 00:51:13,907 INFO ..................................................
2021-12-06 00:51:13,908 INFO [32m'orchestrate_stack' is running from the scenario file: orchestrate_results.yml[0m
2021-12-06 00:51:13,908 INFO ..................................................
2021-12-06 00:51:13,908 INFO Sending out any notifications that are registered.
2021-12-06 00:51:13,909 INFO ..................................................
2021-12-06 00:51:13,909 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:13,910 WARNING ... no tasks to be executed ...
2021-12-06 00:51:13,910 INFO  * Task    : orchestrate
2021-12-06 00:51:13,911 INFO ..................................................
2021-12-06 00:51:13,911 INFO Starting tasks on pipeline: orchestrate
2021-12-06 00:51:13,912 INFO --> Blaster v0.5.0 <--
2021-12-06 00:51:13,912 INFO Task Execution: Sequential
2021-12-06 00:51:13,913 INFO Tasks:
2021-12-06 00:51:13,913 INFO 1. Task     : Install OpenShift
                                Class    : <class 'teflo.tasks.orchestrate.OrchestrateTask'>
                                Methods  : ['run']
2021-12-06 00:51:13,914 INFO ** BLASTER BEGIN **
2021-12-06 00:51:13,915 INFO    running orchestration Install OpenShift for ['scenario_driver']
2021-12-06 00:51:13,916 INFO Executing playbook:
2021-12-06 00:51:13,916 INFO Ansible options used: {}
2021-12-06 00:51:13,916 INFO Executing playbook : ansible/install_ocp.yml
2021-12-06 00:51:13,929 INFO 127.0.0.1
2021-12-06 00:51:14,802 INFO 
2021-12-06 00:51:14,803 INFO PLAY [Install OpenShift] *******************************************************
2021-12-06 00:51:14,810 INFO 
2021-12-06 00:51:14,810 INFO TASK [Gathering Facts] *********************************************************
2021-12-06 00:51:16,397 INFO ok: [127.0.0.1]
2021-12-06 00:51:16,411 INFO 
2021-12-06 00:51:16,411 INFO TASK [Deploy OpenShift cluster] ************************************************
2021-12-06 00:51:16,420 INFO ok: [127.0.0.1] => {
2021-12-06 00:51:16,420 INFO "msg": "Deploying OpenShift using build: None"
2021-12-06 00:51:16,420 INFO }
2021-12-06 00:51:16,427 INFO 
2021-12-06 00:51:16,427 INFO TASK [Write out dummy kubeconfig file] *****************************************
2021-12-06 00:51:17,627 INFO changed: [127.0.0.1]
2021-12-06 00:51:17,634 INFO 
2021-12-06 00:51:17,634 INFO TASK [Ensure archive directory exists] *****************************************
2021-12-06 00:51:18,240 INFO changed: [127.0.0.1]
2021-12-06 00:51:18,302 INFO 
2021-12-06 00:51:18,303 INFO TASK [Archive generated assets] ************************************************
2021-12-06 00:51:18,942 INFO changed: [127.0.0.1]
2021-12-06 00:51:19,011 INFO 
2021-12-06 00:51:19,011 INFO PLAY RECAP *********************************************************************
2021-12-06 00:51:19,011 INFO 127.0.0.1                  : ok=5    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
2021-12-06 00:51:19,011 INFO 
2021-12-06 00:51:19,126 INFO Successfully completed playbook : ansible/install_ocp.yml
2021-12-06 00:51:19,229 INFO Orchestration passed : Successfully completed orchestrate action: Install OpenShift.
2021-12-06 00:51:19,229 INFO ** BLASTER COMPLETE **
2021-12-06 00:51:19,229 INFO     -> TOTAL DURATION: 0h:0m:5s
2021-12-06 00:51:19,230 INFO ..................................................
2021-12-06 00:51:19,230 INFO Sending out any notifications that are registered.
2021-12-06 00:51:19,231 INFO ..................................................
2021-12-06 00:51:19,232 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:19,232 WARNING ... no tasks to be executed ...
2021-12-06 00:51:19,232 INFO ..................................................
2021-12-06 00:51:19,233 INFO [32m'bare_metal_stack' is running from the scenario file: results.yml[0m
2021-12-06 00:51:19,233 INFO ..................................................
2021-12-06 00:51:19,234 INFO Sending out any notifications that are registered.
2021-12-06 00:51:19,234 INFO ..................................................
2021-12-06 00:51:19,235 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:19,235 WARNING ... no tasks to be executed ...
2021-12-06 00:51:19,236 INFO  * Task    : orchestrate
2021-12-06 00:51:19,236 INFO ..................................................
2021-12-06 00:51:19,237 INFO Starting tasks on pipeline: orchestrate
2021-12-06 00:51:19,237 WARNING ... no tasks to be executed ...
2021-12-06 00:51:19,238 INFO ..................................................
2021-12-06 00:51:19,238 INFO Sending out any notifications that are registered.
2021-12-06 00:51:19,239 INFO ..................................................
2021-12-06 00:51:19,239 INFO Starting tasks on pipeline: notify
2021-12-06 00:51:19,239 WARNING ... no tasks to be executed ...
2021-12-06 00:51:19,303 INFO 

2021-12-06 00:51:19,304 INFO                                SCENARIO RUN (END)                              
2021-12-06 00:51:19,304 INFO -------------------------------------------------------------------------------
2021-12-06 00:51:19,304 INFO  * Duration                       : 0h:0m:5s
2021-12-06 00:51:19,305 INFO  * Passed Tasks                   : ['orchestrate']
2021-12-06 00:51:19,305 INFO  * Results Folder                 : /home/jenkins/agent/workspace/PipelineSimulator/sol-mock-openshift4.8/.teflo/.results
2021-12-06 00:51:19,305 INFO  * Iterate Method                 : by_level
2021-12-06 00:51:19,306 INFO  * Scenario Definition            : resource_check
2021-12-06 00:51:19,306 INFO  * Scenario Definition            : provision_localhost
2021-12-06 00:51:19,306 INFO  * Scenario Definition            : orchestrate_stack
2021-12-06 00:51:19,307 INFO  * Scenario Definition            : bare_metal_stack
2021-12-06 00:51:19,307 INFO  * Final Scenario Definition      : /home/jenkins/agent/workspace/PipelineSimulator/sol-mock-openshift4.8/.teflo/.results/results.yml
2021-12-06 00:51:19,307 INFO -------------------------------------------------------------------------------
2021-12-06 00:51:19,308 INFO TEFLO RUN (RESULT=PASSED)
```

**Comment by dno-github** [comment]
Jira ticket CCITCARBON-397 has been created for this issue. Url : https://issues.redhat.com/browse/CCITCARBON-397

**Comment by shay6** [comment]
Working on it in this [PR](https://github.com/RedHatQE/teflo/pull/210)

As you can see below, I'm running a 3 SDF without `notification` block, with `info` level, and with `debug` level.

```
(teflo) shshevac@ bash$~/work/run_teflo$ cat sdf30.yml 
---
include:
  - sdf31.yml
  - sdf32.yml

name: sdf30
description: "Included scenario for py3 environment using linchpin"
provision:
  - name: sdf30host
    groups: local
    ip_address: 127.0.0.1
    ansible_params:
      ansible_connection: local
    labels: ['hello1','hello2']
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ cat sdf31.yml 
---
include:
name: sdf31
description: "Included scenario for py3 environment using linchpin"

provision:
  - name: sdf31host
    groups: local
    ip_address: 127.0.0.1
    ansible_params:
      ansible_connection: local
    labels: ['hello1','hello2']
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ cat sdf32.yml 
---
include:
name: sdf32
description: "Included scenario for py3 environment using linchpin"

provision:
  - name: sdf32host
    groups: local
    ip_address: 127.0.0.1
    ansible_params:
      ansible_connection: local
    labels: ['hello1','hello2']
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ #******************INFO_LEVEL*******************
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ teflo run -s sdf30.yml --log-level info
--------------------------------------------------
Teflo Framework v2.2.3
Copyright (C) 2021, Red Hat, Inc.
--------------------------------------------------
2022-04-11 11:48:01,564 WARNING Scenario workspace was not set, therefore the workspace is automatically assigned to the current working directory. You may experience problems if files needed by teflo do not exists in the scenario workspace.
2022-04-11 11:48:01,591 INFO 

2022-04-11 11:48:01,591 INFO                                TEFLO RUN (START)                               
2022-04-11 11:48:01,591 INFO -------------------------------------------------------------------------------
2022-04-11 11:48:01,592 INFO  * Data Folder           : .teflo/v5qtpm2lbe
2022-04-11 11:48:01,592 INFO  * Workspace             : /home/shshevac/work/run_teflo
2022-04-11 11:48:01,592 INFO  * Log Level             : info
2022-04-11 11:48:01,593 INFO  * Tasks                 : ['validate', 'provision', 'orchestrate', 'execute', 'report', 'cleanup']
2022-04-11 11:48:01,593 INFO  * Iterate Method        : by_level
2022-04-11 11:48:01,594 INFO  * Scenario              : sdf31
2022-04-11 11:48:01,594 INFO  * Scenario              : sdf32
2022-04-11 11:48:01,594 INFO  * Scenario              : sdf30
2022-04-11 11:48:01,595 INFO -------------------------------------------------------------------------------

2022-04-11 11:48:01,595 INFO Validating against all scenarios!
2022-04-11 11:48:01,595 INFO ..................................................
2022-04-11 11:48:01,596 INFO ..................................................
2022-04-11 11:48:01,596 INFO 'sdf31' is validated from the scenario file: sdf31.yml
2022-04-11 11:48:01,597 INFO ..................................................
2022-04-11 11:48:01,597 INFO ..................................................
2022-04-11 11:48:01,598 INFO 'sdf31' is running from the scenario file: sdf31.yml
2022-04-11 11:48:01,598 INFO ..................................................
2022-04-11 11:48:01,600 INFO  * Task    : validate
2022-04-11 11:48:01,601 INFO ..................................................
2022-04-11 11:48:01,601 INFO Starting tasks on pipeline: validate
2022-04-11 11:48:01,602 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,603 INFO Task Execution: Concurrent
2022-04-11 11:48:01,605 INFO Tasks:
2022-04-11 11:48:01,606 INFO 1. Task     : sdf31
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,607 INFO 2. Task     : sdf31host
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,607 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,617 INFO Validating <class 'teflo.resources.scenario.Scenario'> (sdf31)
2022-04-11 11:48:01,618 INFO Validating <class 'teflo.resources.assets.Asset'> (sdf31host)
2022-04-11 11:48:01,663 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,663 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,664 INFO ..................................................
2022-04-11 11:48:01,666 INFO ..................................................
2022-04-11 11:48:01,667 INFO 'sdf32' is validated from the scenario file: sdf32.yml
2022-04-11 11:48:01,667 INFO ..................................................
2022-04-11 11:48:01,668 INFO ..................................................
2022-04-11 11:48:01,668 INFO 'sdf32' is running from the scenario file: sdf32.yml
2022-04-11 11:48:01,668 INFO ..................................................
2022-04-11 11:48:01,671 INFO  * Task    : validate
2022-04-11 11:48:01,671 INFO ..................................................
2022-04-11 11:48:01,672 INFO Starting tasks on pipeline: validate
2022-04-11 11:48:01,672 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,673 INFO Task Execution: Concurrent
2022-04-11 11:48:01,673 INFO Tasks:
2022-04-11 11:48:01,674 INFO 1. Task     : sdf32
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,675 INFO 2. Task     : sdf32host
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,676 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,686 INFO Validating <class 'teflo.resources.scenario.Scenario'> (sdf32)
2022-04-11 11:48:01,687 INFO Validating <class 'teflo.resources.assets.Asset'> (sdf32host)
2022-04-11 11:48:01,732 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,733 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,733 INFO ..................................................
2022-04-11 11:48:01,736 INFO ..................................................
2022-04-11 11:48:01,736 INFO 'sdf30' is validated from the scenario file: sdf30.yml
2022-04-11 11:48:01,736 INFO ..................................................
2022-04-11 11:48:01,737 INFO ..................................................
2022-04-11 11:48:01,737 INFO 'sdf30' is running from the scenario file: sdf30.yml
2022-04-11 11:48:01,738 INFO ..................................................
2022-04-11 11:48:01,740 INFO  * Task    : validate
2022-04-11 11:48:01,741 INFO ..................................................
2022-04-11 11:48:01,741 INFO Starting tasks on pipeline: validate
2022-04-11 11:48:01,741 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,742 INFO Task Execution: Concurrent
2022-04-11 11:48:01,743 INFO Tasks:
2022-04-11 11:48:01,743 INFO 1. Task     : sdf30
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,744 INFO 2. Task     : sdf30host
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,745 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,754 INFO Validating <class 'teflo.resources.scenario.Scenario'> (sdf30)
2022-04-11 11:48:01,755 INFO Validating <class 'teflo.resources.assets.Asset'> (sdf30host)
2022-04-11 11:48:01,801 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,801 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,802 INFO ..................................................
2022-04-11 11:48:01,804 INFO ..................................................
2022-04-11 11:48:01,805 INFO 'sdf31' is running from the scenario file: sdf31.yml
2022-04-11 11:48:01,805 INFO ..................................................
2022-04-11 11:48:01,807 INFO  * Task    : provision
2022-04-11 11:48:01,808 INFO ..................................................
2022-04-11 11:48:01,808 INFO Starting tasks on pipeline: provision
2022-04-11 11:48:01,808 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,809 INFO Task Execution: Concurrent
2022-04-11 11:48:01,810 INFO Tasks:
2022-04-11 11:48:01,810 INFO 1. Task     : sdf31host
                                Class    : <class 'teflo.tasks.provision.ProvisionTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,811 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,819 WARNING Asset sdf31host is static, provision will be skipped.
2022-04-11 11:48:01,821 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,821 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,822 INFO Populating inventory file with host(s) sdf31host
2022-04-11 11:48:01,824 INFO ..................................................
2022-04-11 11:48:01,825 INFO  * Task    : orchestrate
2022-04-11 11:48:01,826 INFO ..................................................
2022-04-11 11:48:01,826 INFO Starting tasks on pipeline: orchestrate
2022-04-11 11:48:01,826 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,827 INFO ..................................................
2022-04-11 11:48:01,827 INFO  * Task    : execute
2022-04-11 11:48:01,828 INFO ..................................................
2022-04-11 11:48:01,828 INFO Starting tasks on pipeline: execute
2022-04-11 11:48:01,829 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,829 INFO ..................................................
2022-04-11 11:48:01,830 INFO  * Task    : report
2022-04-11 11:48:01,831 INFO ..................................................
2022-04-11 11:48:01,831 INFO Starting tasks on pipeline: report
2022-04-11 11:48:01,831 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,832 INFO ..................................................
2022-04-11 11:48:01,834 INFO ..................................................
2022-04-11 11:48:01,834 INFO 'sdf32' is running from the scenario file: sdf32.yml
2022-04-11 11:48:01,835 INFO ..................................................
2022-04-11 11:48:01,837 INFO  * Task    : provision
2022-04-11 11:48:01,837 INFO ..................................................
2022-04-11 11:48:01,838 INFO Starting tasks on pipeline: provision
2022-04-11 11:48:01,838 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,839 INFO Task Execution: Concurrent
2022-04-11 11:48:01,839 INFO Tasks:
2022-04-11 11:48:01,840 INFO 1. Task     : sdf32host
                                Class    : <class 'teflo.tasks.provision.ProvisionTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,841 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,849 WARNING Asset sdf32host is static, provision will be skipped.
2022-04-11 11:48:01,852 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,852 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,853 INFO Populating inventory file with host(s) sdf32host
2022-04-11 11:48:01,868 INFO ..................................................
2022-04-11 11:48:01,869 INFO  * Task    : orchestrate
2022-04-11 11:48:01,872 INFO ..................................................
2022-04-11 11:48:01,874 INFO Starting tasks on pipeline: orchestrate
2022-04-11 11:48:01,876 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,877 INFO ..................................................
2022-04-11 11:48:01,878 INFO  * Task    : execute
2022-04-11 11:48:01,881 INFO ..................................................
2022-04-11 11:48:01,882 INFO Starting tasks on pipeline: execute
2022-04-11 11:48:01,883 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,883 INFO ..................................................
2022-04-11 11:48:01,884 INFO  * Task    : report
2022-04-11 11:48:01,886 INFO ..................................................
2022-04-11 11:48:01,886 INFO Starting tasks on pipeline: report
2022-04-11 11:48:01,887 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,888 INFO ..................................................
2022-04-11 11:48:01,891 INFO ..................................................
2022-04-11 11:48:01,892 INFO 'sdf30' is running from the scenario file: sdf30.yml
2022-04-11 11:48:01,892 INFO ..................................................
2022-04-11 11:48:01,895 INFO  * Task    : provision
2022-04-11 11:48:01,896 INFO ..................................................
2022-04-11 11:48:01,897 INFO Starting tasks on pipeline: provision
2022-04-11 11:48:01,897 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,898 INFO Task Execution: Concurrent
2022-04-11 11:48:01,899 INFO Tasks:
2022-04-11 11:48:01,900 INFO 1. Task     : sdf30host
                                Class    : <class 'teflo.tasks.provision.ProvisionTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,901 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,909 WARNING Asset sdf30host is static, provision will be skipped.
2022-04-11 11:48:01,911 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,911 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,912 INFO Populating inventory file with host(s) sdf30host
2022-04-11 11:48:01,914 INFO ..................................................
2022-04-11 11:48:01,915 INFO  * Task    : orchestrate
2022-04-11 11:48:01,916 INFO ..................................................
2022-04-11 11:48:01,916 INFO Starting tasks on pipeline: orchestrate
2022-04-11 11:48:01,916 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,917 INFO ..................................................
2022-04-11 11:48:01,917 INFO  * Task    : execute
2022-04-11 11:48:01,918 INFO ..................................................
2022-04-11 11:48:01,918 INFO Starting tasks on pipeline: execute
2022-04-11 11:48:01,919 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,919 INFO ..................................................
2022-04-11 11:48:01,919 INFO  * Task    : report
2022-04-11 11:48:01,920 INFO ..................................................
2022-04-11 11:48:01,920 INFO Starting tasks on pipeline: report
2022-04-11 11:48:01,921 WARNING ... no tasks to be executed ...
2022-04-11 11:48:01,921 INFO ..................................................
2022-04-11 11:48:01,923 INFO ..................................................
2022-04-11 11:48:01,924 INFO 'sdf30' is running from the scenario file: sdf30.yml
2022-04-11 11:48:01,924 INFO ..................................................
2022-04-11 11:48:01,925 INFO ..................................................
2022-04-11 11:48:01,925 INFO 'sdf30' is running from the scenario file: sdf30.yml
2022-04-11 11:48:01,926 INFO ..................................................
2022-04-11 11:48:01,928 INFO  * Task    : cleanup
2022-04-11 11:48:01,929 INFO ..................................................
2022-04-11 11:48:01,930 INFO Starting tasks on pipeline: cleanup
2022-04-11 11:48:01,930 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,931 INFO Task Execution: Sequential
2022-04-11 11:48:01,931 INFO Tasks:
2022-04-11 11:48:01,932 INFO 1. Task     : sdf30host
                                Class    : <class 'teflo.tasks.cleanup.CleanupTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,932 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,934 INFO    cleanup asset sdf30host
2022-04-11 11:48:01,934 WARNING Asset sdf30host is static, skipping teardown.
2022-04-11 11:48:01,935 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,935 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,936 INFO ..................................................
2022-04-11 11:48:01,938 INFO ..................................................
2022-04-11 11:48:01,938 INFO 'sdf32' is running from the scenario file: sdf32.yml
2022-04-11 11:48:01,939 INFO ..................................................
2022-04-11 11:48:01,939 INFO ..................................................
2022-04-11 11:48:01,940 INFO 'sdf32' is running from the scenario file: sdf32.yml
2022-04-11 11:48:01,940 INFO ..................................................
2022-04-11 11:48:01,942 INFO  * Task    : cleanup
2022-04-11 11:48:01,943 INFO ..................................................
2022-04-11 11:48:01,944 INFO Starting tasks on pipeline: cleanup
2022-04-11 11:48:01,944 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,945 INFO Task Execution: Sequential
2022-04-11 11:48:01,945 INFO Tasks:
2022-04-11 11:48:01,945 INFO 1. Task     : sdf32host
                                Class    : <class 'teflo.tasks.cleanup.CleanupTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,946 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,947 INFO    cleanup asset sdf32host
2022-04-11 11:48:01,948 WARNING Asset sdf32host is static, skipping teardown.
2022-04-11 11:48:01,948 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,949 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,949 INFO ..................................................
2022-04-11 11:48:01,951 INFO ..................................................
2022-04-11 11:48:01,952 INFO 'sdf31' is running from the scenario file: sdf31.yml
2022-04-11 11:48:01,952 INFO ..................................................
2022-04-11 11:48:01,953 INFO ..................................................
2022-04-11 11:48:01,953 INFO 'sdf31' is running from the scenario file: sdf31.yml
2022-04-11 11:48:01,954 INFO ..................................................
2022-04-11 11:48:01,956 INFO  * Task    : cleanup
2022-04-11 11:48:01,957 INFO ..................................................
2022-04-11 11:48:01,958 INFO Starting tasks on pipeline: cleanup
2022-04-11 11:48:01,958 INFO --> Blaster v0.5.0 <--
2022-04-11 11:48:01,958 INFO Task Execution: Sequential
2022-04-11 11:48:01,959 INFO Tasks:
2022-04-11 11:48:01,959 INFO 1. Task     : sdf31host
                                Class    : <class 'teflo.tasks.cleanup.CleanupTask'>
                                Methods  : ['run']
2022-04-11 11:48:01,960 INFO ** BLASTER BEGIN **
2022-04-11 11:48:01,961 INFO    cleanup asset sdf31host
2022-04-11 11:48:01,962 WARNING Asset sdf31host is static, skipping teardown.
2022-04-11 11:48:01,963 INFO ** BLASTER COMPLETE **
2022-04-11 11:48:01,963 INFO     -> TOTAL DURATION: 0h:0m:0s
2022-04-11 11:48:01,964 INFO ..................................................
2022-04-11 11:48:01,979 INFO 

2022-04-11 11:48:01,980 INFO                                SCENARIO RUN (END)                              
2022-04-11 11:48:01,980 INFO -------------------------------------------------------------------------------
2022-04-11 11:48:01,980 INFO  * Duration                       : 0h:0m:0s
2022-04-11 11:48:01,981 INFO  * Passed Tasks                   : ['validate', 'provision', 'orchestrate', 'execute', 'report', 'cleanup']
2022-04-11 11:48:01,981 INFO  * Results Folder                 : .teflo/.results
2022-04-11 11:48:01,981 INFO  * Iterate Method                 : by_level
2022-04-11 11:48:01,982 INFO  * Scenario Definition            : sdf31
2022-04-11 11:48:01,982 INFO  * Scenario Definition            : sdf32
2022-04-11 11:48:01,983 INFO  * Scenario Definition            : sdf30
2022-04-11 11:48:01,983 INFO  * Final Scenario Definition      : .teflo/.results/results.yml
2022-04-11 11:48:01,983 INFO -------------------------------------------------------------------------------
2022-04-11 11:48:01,984 INFO TEFLO RUN (RESULT=PASSED)
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ #******************DEBUG_LEVEL*******************
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ teflo run -s sdf30.yml --log-level debug 2>&1 | grep -e "Starting tasks on pipeline: notify" 
2022-04-11 11:53:30,239 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,306 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,311 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,373 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,378 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,442 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,445 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,472 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,475 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,504 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,507 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,534 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,540 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,550 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,555 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,564 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,569 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
2022-04-11 11:53:30,579 DEBUG [teflo.teflo._run_pipeline:698] Starting tasks on pipeline: notify
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 
(teflo) shshevac@ bash$~/work/run_teflo$ 




```

**Comment by shay6** [comment]
Hey @ryankwilliams,

Can you please check this issue with my branch [1]  to see if it solved your problem? Thanks

[1] https://github.com/shay6/teflo/tree/tkt_397

**Comment by ryankwilliams** [comment]
Hey @shay6, I am currently on parental leave and I am unable to try out this patch of teflo from your branch. I reviewed the output you provided and it looks much better (when running with info level logging - removing the statements regarding notify when none are provided). Thanks!

**Comment by shay6** [comment]
Thanks, @ryankwilliams, for the response. Enjoy in your parental leave!

**Comment by rujutashinde** [comment]
fixed with #210
